source/beamline/BioXAS/BioXASMainM1MirrorBendControl.cpp

```diff
@@ -0,0 +1,42 @@
+#include "BioXASMainM1MirrorBendControl.h"
+
+BioXASMainM1MirrorBendControl::BioXASMainM1MirrorBendControl(const QString &name, const QString &units, QObject *parent, const QString &description) :
+	BioXASMirrorBendControl(name, units, parent, description)
+{
+	/// Initialize inherited variables.
+
+	setMinimumValue( calculateBendRadius(12, 12) );
+	setMaximumValue( calculateBendRadius(0.0001, 0.0001) );
+	setTolerance(50);
+
+	/// Current settings.
+
+	updateStates();
+}
+
+BioXASMainM1MirrorBendControl::~BioXASMainM1MirrorBendControl()
+{
+
+}
+
+double BioXASMainM1MirrorBendControl::calculateBendRadius(double upstreamBenderValue, double downstreamBenderValue)
+{
+	double radius1 = -7334.49 + 110164/upstreamBenderValue + 1908.99 * log(upstreamBenderValue);
```

Comment:
The ```110164/upstreamBenderValue``` part of this equation is okay, but I just wanted to flag this because if upstreamBeanderValue was also an integer you'd be undergoing integer division which is _not_ what you want.  Just something to keep in mind.